src/main/java/edu/harvard/iq/dataverse/search/IndexServiceBean.java

```diff
@@ -402,7 +445,16 @@ public void asyncIndexDataset(Dataset dataset, boolean doNormalSolrDocCleanUp) {
     @Asynchronous
     public void asyncIndexDatasetList(List<Dataset> datasets, boolean doNormalSolrDocCleanUp) {
         for(Dataset dataset : datasets) {
-            asyncIndexDataset(dataset, true);
+            try {
+                ASYNC_INDEX_SEMAPHORE.acquire();
```

Comment:
```suggestion
               acquirePermitFromSemaphore();
```

I forgot to fix this one!

Reply:
I just added it and committed.